src/middlewares/setUserId.middleware.ts

```diff
@@ -0,0 +1,26 @@
+import { NextFunction, Request, RequestHandler, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { CustomError } from './errors';
+import { UNAUTHORIZED } from 'http-status';
+
+declare global {
+  namespace Express {
+    interface Request {
+      userId: number;
+    }
+  }
+}
+
+const setUserId: RequestHandler = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const { token } = req.cookies;
+  if (!token) next(new CustomError('Unauthenticated', UNAUTHORIZED));
```

Comment:
A better practice to import it like: `import HttpStatus from 'http-status';`. So when anyone read it, he can easily know what it means without going to the top.